src/main/java/com/hazelcast/aws/impl/DescribeInstances.java

```diff
@@ -76,30 +73,48 @@ public DescribeInstances(AwsConfig awsConfig, String endpoint) throws IOExceptio
         addFilters();
     }
 
+    //Just for testing purposes
     DescribeInstances(AwsConfig awsConfig) {
-        if (awsConfig == null) {
-            throw new IllegalArgumentException("AwsConfig is required!");
-        }
         this.awsConfig = awsConfig;
     }
 
-    void checkKeysFromIamRoles(Environment env) throws IOException {
-        if (awsConfig.getAccessKey() != null && awsConfig.getIamRole() == null) {

```

Comment:
If accessKey is set, we used to use it but not anymore.. That might break compatibility?